### ISSUE TITLE
remove selectedRow in handsontable using jQuery

### ISSUE THREAD

**Issue by chethan1095** [handsontable/handsontable#5693]
![table1](https://user-images.githubusercontent.com/33021783/50478879-acac7200-09f9-11e9-9615-8778538eacad.PNG)
 I

> I have an handsontable like this after selecting any one of the row (for Ex: LastName) it should remove that particular row permanently how to achieve it using jquery

**This is my code i have written afterSelection function but i don't know how to delete that row**

`   onShow:function(){
        this.tab.handsontable({
        data: self.data,
        dataSchema: { columnsexpo: null, placeholder: null },
        colHeaders: ['Columns Export'],
        rowHeaders: true,
        fixedColumns: true,
        fillHandle: {
            autoInsertRow: false,
        },
        columnSorting: true,
        columns: [
          { data: 'columnsexpo', readOnly: true },
        ],
       // contextMenu: ["remove_row"],
        stretchH: 'all',
        className: "htCenter",
        height: 420,
        afterChange: function () {

        },
        beforeRemoveRow: function (row, col) {
            var m = this.getDataAtCell(row, 0);
            var mandatory = true;
            self.MandatoryFields.forEach(function (item) {
                if (!_.isEmpty(m)) {
                    var found = m.toLowerCase().includes(item.toLowerCase());
                    if (found) {
                        mandatory = false;
                    }
                }
            });
            if (!mandatory) {
                return false
            } else
                return true;
        },
        afterSelection: function (r, c) {
            var da = this.getDataAtRow(r);
            selectedRow = "";
            selectedRow = da[0];                
            console.log(selectedRow);             

        },            
        afterRender: function () {
            if (init) {
                Events.trigger("DEW:ValidRequest", 1, self.checkValid());
            }
            init = true;
            $('#tablesortable thead th div').filter(function () { return $(this).text() == "Columns Export"; }).popup({
                title: 'Columns Export',
                position: 'top center'
            });               
        }
    });
  }`

**Comment by AMBudnik** [comment]
hey @chethan1095 

you can use [alter method.](https://handsontable.com/docs/6.2.2/Core.html#alter) using your `selectedRow` variable.

**Comment by chethan1095** [comment]
sorry i don't know how that works can u see my code and answer it please it will help me lot

**Comment by AMBudnik** [comment]
You can try `this.alter('remove_row', selectedRow, 1)` and then repaint the table` this.render()`

**Comment by chethan1095** [comment]
where i should use this in the above code @AMBudnik

**Comment by chethan1095** [comment]
`this.alter('remove_row', selectedRow, 1)` is not working

**Comment by AMBudnik** [comment]
Can you push your settings to a demo? Maybe we're missing something.

**Comment by chethan1095** [comment]
> check my code now please

**Comment by chethan1095** [comment]
Thank you for this but i have some small issue i am able to remove the row data but its not removing selected row instead it is removing last row

**Comment by AMBudnik** [comment]
did you check (logged in the console or set a debugger) for the row index?

**Comment by chethan1095** [comment]
Hey @AMBudnik i am able to remove selected row but when i select last row it is removing all previous selected row please help me this

**Comment by AMBudnik** [comment]
My demo deletes the row you selected - please observe the data you remove, when you click A5 the row with this data disappears.

**Comment by AMBudnik** [comment]
If that is not the desired effect and you could share a draft/drawing please send it to me via support@handsontable.com

**Comment by chethan1095** [comment]
yes the row with selected data will remove but select last row in that it will; remove all other rows

**Comment by chethan1095** [comment]
hey @AMBudnik please check i had sent you an mail to support@handsontable.com
